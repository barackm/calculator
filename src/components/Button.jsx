import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, onClick }) => (
  <button type="button" onClick={() => onClick(name)}>
    {name}
  </button>
);

Button.propTypes = {
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default Button;
