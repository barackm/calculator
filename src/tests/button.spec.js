import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('Button Component', () => {
  it('Should match spanshot', () => {
    const tree = renderer
      .create(<Button name="1" onClick={() => {}} color="white" wide />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
