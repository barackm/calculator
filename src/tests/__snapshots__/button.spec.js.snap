// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should match spanshot 1`] = `
<button
  className="zero"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
    }
  }
  type="button"
>
  1
</button>
`;
